flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/relational/connection/JdbcConnectionPools.java

```diff
@@ -50,9 +53,19 @@ public HikariDataSource getOrCreateConnectionPool(
         synchronized (pools) {
             if (!pools.containsKey(poolId)) {
                 LOG.info("Create and register connection pool {}", poolId);
-                pools.put(poolId, jdbcConnectionPoolFactory.createPooledDataSource(sourceConfig));
+                pools.put(
+                        poolId,
+                        poolFactoryMap
+                                .get(poolId.getDataSourcePoolFactoryIdentifier())
```

Comment:
what would happen if  the `poolFactoryMap` doesn't contain the given `dataSourcePoolFactoryIdentifier`?